elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala

```diff
@@ -1206,16 +1206,24 @@ trait RestlasticSearchClientTest {
       val count = Await.result(restClient.count(index, tpe, new QueryRoot(MatchAll)), 10.seconds)
       count should be(docsCount)
 
-      val termQuery = TermQuery("text7", "here7")
-
       val delFut = restClient.deleteByQuery(index, tpe, new QueryRoot(termQuery), true)
-      Await.result(delFut, 20.seconds)
+      val deleteDocsResponse = Await.result(delFut, 20.seconds)
+      deleteDocsResponse.deletedDocumentsCount should be(docsCount)
       refresh()
+    }
+
+    "Support deleting by a query" in {
+      val termQuery = TermQuery("text7", "here7")
+      checkDeletingByAQueryResponse(termQuery)
 
       val count1 = Await.result(restClient.count(index, tpe, new QueryRoot(termQuery)), 10.seconds)
       count1 should be(0)
     }
 
+    "Deleting by a query without waiting generates correct response" in {
+      checkDeletingByAQueryResponse(TermQuery("text7", "here7"))
```

Comment:
Here in this test we want to test 
`val delFut = restClient.deleteByQuery(index, tpe, new QueryRoot(termQuery), false)`

Reply:
I suggest to copy/paste the test case, generally the code duplication is bad, but in the unit tests I think It's better to duplicate the code to make the testing scenario as clear as possible (and make every tests independent).

Reply:
And I'm afraid in this case the returned deleted could could be `-1`.

Reply:
Which is correct for the asynchronous way.

Reply:
Fixed